pkg/orchestrator/controller/controller.go

```diff
@@ -219,28 +238,39 @@ func (c *controller) OnRemove(clusterID string) {
 }
 
 func runQueueWorker(clusterID string, federator federate.Federator, informer cache.SharedIndexInformer,
-	workQueue workqueue.RateLimitingInterface) {
+	workQueue workqueue.RateLimitingInterface, resourceKeyProvider resourceKeyProviderFunc) {
 
 	for {
-		key, shutdown := workQueue.Get()
+		keyObj, shutdown := workQueue.Get()
 		if shutdown {
 			klog.V(2).Infof("Submariner watcher for cluster \"%s\" stopped", clusterID)
 			return
 		}
 
+		key := keyObj.(string)
+
 		klog.V(2).Infof("Processing key \"%s\" for cluster \"%s\"", key, clusterID)
 
 		func() {
 			defer workQueue.Done(key)
 
-			item, exists, err := informer.GetIndexer().GetByKey(key.(string))
+			item, exists, err := informer.GetIndexer().GetByKey(key)
 			if err != nil {
 				klog.Errorf("Error fetching object with key %s from store: %v", key, err)
 				workQueue.AddRateLimited(key)
 			} else if !exists {
-				klog.V(2).Infof("Key \"%s\" does not exist - deleting distributed resource", key)
-				// TODO - handle delete
-				workQueue.Forget(key)
+				// Ignoring the error as it should never happen as we're using the corresponding function(s) to create the keys.
+				namespace, name, _ := cache.SplitMetaNamespaceKey(key)
+				resourceKey := resourceKeyProvider(namespace, name)
+
+				klog.V(2).Infof("Key \"%s\" does not exist - deleting distributed resource: %#v", key, resourceKey)
```

Comment:
instead of `\"%s\"` use `%q`

Reply:
oh yeah - forgot about that - changed other places too to be consistent.